62.CAD/vertex_shader.hlsl

```diff
@@ -398,20 +445,68 @@ PSInput main(uint vertexID : SV_VertexID)
+    {
+        outV.setColor(lineStyle.color);
+        outV.setLineThickness(screenSpaceLineWidth / 2.0f);
+
+        CurveBox curveBox;
+        curveBox.aabbMin = vk::RawBufferLoad<double2>(drawObj.geometryAddress, 8u);
+        curveBox.aabbMax = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2), 8u);
+        curveBox.curveMin[0] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 2, 8u);
+        curveBox.curveMin[1] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 3, 8u);
+        curveBox.curveMin[2] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 4, 8u);
+        curveBox.curveMax[0] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 5, 8u);
+        curveBox.curveMax[1] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 6, 8u);
+        curveBox.curveMax[2] = vk::RawBufferLoad<double2>(drawObj.geometryAddress + sizeof(double2) * 7, 8u);
+
+        const bool2 maxCorner = bool2(vertexIdx & 0x1u, vertexIdx >> 1);
+        const double2 coord = transformPointNdc(lerp(curveBox.aabbMin, curveBox.aabbMax, maxCorner));
+        outV.position = float4((float2) coord, 0.f, 1.f);
+
+        const uint major = 1;
+        const uint minor = 1-major;
+
+        nbl::hlsl::shapes::QuadraticBezier curveMin = nbl::hlsl::shapes::QuadraticBezier::construct(
+            curveBox.curveMin[0], curveBox.curveMin[1], curveBox.curveMin[2]);
+        nbl::hlsl::shapes::QuadraticBezier curveMax = nbl::hlsl::shapes::QuadraticBezier::construct(
+            curveBox.curveMax[0], curveBox.curveMax[1], curveBox.curveMax[2]);
+
+        // Swizzled X = major
+        // Swizzled Y = minor
+        float2 uv = (float2) (major == 1 ? maxCorner.yx : maxCorner.xy);
+        outV.setUVMinor(uv.y);
+        outV.setCurveMinA(major == 1 ? curveMin.A().yx : curveMin.A().xy);
+        outV.setCurveMinB(major == 1 ? curveMin.B().yx : curveMin.B().xy);
+        outV.setCurveMinC(major == 1 ? curveMin.C().yx : curveMin.C().xy);
+        outV.setCurveMaxA(major == 1 ? curveMax.A().yx : curveMax.A().xy);
+        outV.setCurveMaxB(major == 1 ? curveMax.B().yx : curveMax.B().xy);
+        outV.setCurveMaxC(major == 1 ? curveMax.C().yx : curveMax.C().xy);
+
+        {
+            float a = outV.getCurveMinA().x;
+            float b = outV.getCurveMinB().x;
+            float c = outV.getCurveMinC().x - uv.x;
+
+            float det = b*b-4.f*a*c;
+            float rcp = 0.5f/a;
+            outV.setMinCurveDetRcp2(det * rcp * rcp);
+            outV.setMinCurveBrcp(b * rcp);
+        }
+        {
+            float a = outV.getCurveMaxA().x;
+            float b = outV.getCurveMaxB().x;
+            float c = outV.getCurveMaxC().x - uv.x;
+
+            float det = b*b-4.f*a*c;
+            float rcp = 0.5f/a;
+            outV.setMaxCurveDetRcp2(det * rcp * rcp);
+            outV.setMaxCurveBrcp(b * rcp);
```

Comment:
this repeated thing could be refactored into a function that gets A,B,C values of both curves and sets internal values, then you can pass set that struct

here is what I expect it to look like eventually
```c++
nbl::hlsl::equations::Quadratic minQuadratic = nbl::hlsl::equations::Quadratic(aMin,bMin,cMin);
nbl::hlsl::equations::Quadratic maxQuadratic = nbl::hlsl::equations::Quadratic(aMax,bMax,cMax);
outV.setMinCurve(minCurve);
outV.setMaxCurve(maxCurve);
minQuadratic.c -= minorNDC;
maxQuadratic.c -= minorNDC;
nbl::hlsl::equations::Quadratic::RootFinder minCurveRootFinder = nbl::hlsl::equations::Quadratic::RootFinder(minQuadratic); // this does all the computations and stores detRcp2 and Brcp
nbl::hlsl::equations::Quadratic::PrecomputedRootFinder maxCurveRootFinder = nbl::hlsl::equations::Quadratic::PrecomputedRootFinder (maxQuadratic); // this does all the computations and stores detRcp2 and Brcp
outV.setCurveBoxRootFinders(minCurveRootFinder, maxCurveRootFinder);  // this sets the structs detRcp2 and Brcp to it's internal interp and non interp data
```

Notice that I'm using equations::Quadratic and not shapes::Quadratic cause you'll have to store `float`s instead of `float2`s. (you'll have to create a new hlsl file and namespace + add to cmake)

which indicates that you have to move your functions related to solving quadratic equations to that builtin function

Get Ideas by looking at @Przemog1's PR on shapes::Quadratic and shapes::Quadratic::AnalyticArcLengthCalculator and looking at how he uses it in his examples PR

Reply:
I thought @devshgraphicsprogramming said the precomputed root finding stuff and so on was too specific to this and wouldn't be going into the HLSL include stuff?

Reply:
i'll leave it in common.hlsl for now

Reply:
> I thought @devshgraphicsprogramming said the precomputed root finding stuff and so on was too specific to this and wouldn't be going into the HLSL include stuff?

I think I said that xD
But I changed my mind